functests/1_performance/netqueues.go

```diff
@@ -0,0 +1,359 @@
+				}
+
+				for _, node := range workerRTNodes {
+					vid = getVendorID(node, device)
+					did = getDeviceID(node, device)
+				}
+				if profile.Spec.Net.UserLevelNetworking != nil && *profile.Spec.Net.UserLevelNetworking && len(profile.Spec.Net.Devices) != 0 {
+					By("Enable UserLevelNetworking and add DeviceID, VendorID and Interface in Profile")
+					profile.Spec.Net = &performancev2.Net{
+						UserLevelNetworking: &net,
+						Devices: []performancev2.Device{
+							{
+								InterfaceName: &device,
+							},
+							{
+								VendorID: &vid,
+								DeviceID: &did,
+							},
+						},
+					}
+					profiles.UpdateWithRetry(profile)
+					Eventually(func() bool {
+						err := checkDeviceSupport(workerRTNodes, devices)
+						Expect(err).ToNot(HaveOccurred())
+						return true
+					}, cluster.ComputeTestTimeout(240*time.Second, RunningOnSingleNode), testPollInterval*time.Second).Should(BeTrue())
+				}
+				for _, size := range devices {
+					if size == getReservedCPUSize(profile.Spec.CPU) {
+						count++
+					}
+				}
+				if count == 0 {
+					Skip("Skipping Test: Unable to set Network queue size to reserved cpu count")
+				}
+				//Verify the tuned profile is created on the worker-cnf nodes:
+				tunedCmd := []string{"bash", "-c",
+					fmt.Sprintf("cat /etc/tuned/openshift-node-performance-performance/tuned.conf | grep devices_udev_regex")}
+				for _, node := range workerRTNodes {
+					tunedPod := getTunedPod(&node)
+					out, err := pods.ExecCommandOnPod(tunedPod, tunedCmd)
+					deviceExists := strings.ContainsAny(string(out), device)
+					Expect(deviceExists).To(BeTrue())
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}
+		})
+	})
+})
+
+//Check if the device support multiple queues
+func checkDeviceSupport(workernodes []corev1.Node, devices map[string]int) error {
+	cmdGetPhysicalDevices := []string{"find", "/sys/class/net", "-type", "l", "-not", "-lname", "*virtual*", "-printf", "%f "}
+	var channelCurrentCombined int
+	var err error
+	for _, node := range workernodes {
+		tunedPod := getTunedPod(&node)
+		phyDevs, err := pods.ExecCommandOnPod(tunedPod, cmdGetPhysicalDevices)
+		Expect(err).ToNot(HaveOccurred())
+		for _, d := range strings.Split(string(phyDevs), " ") {
+			if d == "" {
+				continue
+			}
+			_, err := pods.ExecCommandOnPod(tunedPod, []string{"ethtool", "-l", d})
+			if err == nil {
+				cmdCombinedChannelsCurrent := []string{"bash", "-c",
+					fmt.Sprintf("ethtool -l %s | sed -n '/Current hardware settings:/,/Combined:/{s/^Combined:\\s*//p}'", d)}
+				out, err := pods.ExecCommandOnPod(tunedPod, cmdCombinedChannelsCurrent)
+				if strings.Contains(string(out), "n/a") {
+					fmt.Printf("Device %s doesn't support multiple queues\n", d)
+				} else {
+					channelCurrentCombined, err = strconv.Atoi(strings.TrimSpace(string(out)))
+					if err != nil {
+						testlog.Warningf(fmt.Sprintf("unable to retrieve current multi-purpose channels hardware settings for device %s on %s",
```

Comment:
I wonder if the warning is enough here,  this will cause the test **not** to fail on flakiness of getting the channels but would exclude an actual check for a specific device in some test.

Reply:
@yanirq i am not sure how to address this because since these tests are added in both upstream and down stream. And in upstream the support of supported netqueue devices is something not guaranteed. The tests always fails if i fail it.

Reply:
ack . so lets leave the warning